onmt/dynamic/train.py

```diff
@@ -25,6 +29,22 @@ def train(opt):
 
     set_random_seed(opt.seed, False)
 
+    DynamicArgumentParser.get_all_transform(opt)
+
+    transforms_cls = get_transforms_cls(opt._all_transform)
+    specials = get_specials(opt, transforms_cls)
+
+    print(opt)
+
+    fields = build_dynamic_fields(
+        opt, src_specials=specials['src'], tgt_specials=specials['tgt'])
+    save_fields(opt, fields)
+
+    transforms = make_transforms(opt, transforms_cls, fields)
+    save_transforms(opt, transforms)
+    if opt.verbose:
+        save_transformed_sample(opt, transforms)
+
```

Comment:
Maybe create a separate function to do those stuff could make the intention more explicit?

Reply:
Absolutely. This is a very first draft of the merge.